frontend/packages/data-portal/app/graphql/getDatasetsFilterData.server.ts

```diff
@@ -0,0 +1,57 @@
+import type { ApolloClient, NormalizedCacheObject } from '@apollo/client'
+
+import { gql } from 'app/__generated__'
+import { Datasets_Bool_Exp } from 'app/__generated__/graphql'
+
+const GET_DATASETS_FILTER_DATA_QUERY = gql(`
```

Comment:
Is it possible to just make this a fragment? https://www.apollographql.com/docs/react/data/fragments/ That way any query can add this just by concatenating the string. It should generate the type too, hopefully that would work with `useTypedLoaderData()`?

```
const TEST_FRAGMENT = gql(`
  fragment TestFragment on query_root {
    datasets {
      organism_name
    }
  }  
`)
```

```
useTypedLoaderData<TestFragmentFragment>()
```

Reply:
yes i think so. a lot of the codebase could actually benefit from fragments (for example authors)

how important do you think this is? since i have low bandwidth i would prefer to keep this as-is and make the refactor along with the other places we should be using fragments

Reply:
I don't think it's too important! Maybe file a ticket?